plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoPrivileges.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.mongodb;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoDatabase;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.QueryRunner;
+import org.bson.BsonArray;
+import org.bson.BsonDocument;
+import org.bson.BsonString;
+import org.testng.annotations.Test;
+
+import java.util.Locale;
+import java.util.Optional;
+
+import static io.airlift.testing.Closeables.closeAllSuppress;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestMongoPrivileges
+        extends AbstractTestQueryFramework
+{
+    private static final String ROOT_USER = "root";
+    private static final String ROOT_PASSWORD = "password";
+    private static final String TEST_USER = "testUser";
+    private static final String TEST_PASSWORD = "pass";
+    private static final String TEST_ROLE = "testRole";
+    private static final String TEST_DATABASE = "test";
+    private static final String TEST_COLLECTION = "testCollection";
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        MongoServer mongoServer = closeAfterClass(setupMongoServer());
+        return createMongoQueryRunner(mongoServer.getConnectionString(TEST_USER, TEST_PASSWORD).toString() + "/?authSource=" + TEST_DATABASE);
+    }
+
+    @Test
+    public void testSchemasVisibility()
+    {
+        assertThat(getQueryRunner().execute("SHOW SCHEMAS FROM mongodb")
+                .getOnlyColumnAsSet())
+                .containsExactly("information_schema", TEST_DATABASE);
+    }
+
+    @Test
+    public void testTablesVisibility()
+    {
+        assertThat(getQueryRunner().execute("SHOW TABLES FROM mongodb." + TEST_DATABASE)
+                .getOnlyColumnAsSet())
+                .containsExactly(TEST_COLLECTION.toLowerCase(Locale.ENGLISH));
```

Comment:
Do we need a redundant conversion ? Can we use `assertQuery` where we could provide a result as `VALUES ...`

Reply:
Replaced